sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatThreadClientBuilder.java

```diff
@@ -114,10 +104,26 @@ public ChatThreadClientBuilder addPolicy(HttpPipelinePolicy pipelinePolicy) {
         return this;
     }
 
+    /**
+     * Sets the {@link RetryPolicy} that will attempt to retry failed requests, if applicable.
+     *
+     * @param retryPolicy the retryPolicy value.
+     * @throws NullPointerException if retryPolicy is null.
+     * @return The updated {@link ChatThreadClientBuilder} object.
+     */
+    public ChatThreadClientBuilder retryPolicy(RetryPolicy retryPolicy) {
+        if (retryPolicy == null) {
+            throw logger.logExceptionAsError(new NullPointerException("retryPolicy is required."));
+        }
+        this.retryPolicy = retryPolicy;
+        return this;
+    }
+
     /**
      * Sets the {@link HttpLogOptions} for service requests.
      *
      * @param logOptions The logging configuration to use when sending and receiving HTTP requests/responses.
+     * @throws NullPointerException if logOptions is null.
      * @return The updated {@link ChatThreadClientBuilder} object
      */
     public ChatThreadClientBuilder httpLogOptions(HttpLogOptions logOptions) {
```

Comment:
`logOptions` can be `null` if we have a default value to set it to when not present, [which we do](https://github.com/Azure/azure-sdk-for-android/blob/209ef5b97c5f2bb7a127db07febeee68bec1a0e6/sdk/core/azure-core-http/src/main/java/com/azure/android/core/http/policy/HttpLoggingPolicy.java#L60).

Reply:
Same as retry policy. Here we only want to check when user calls httpLogOptions(HttpLogOptions logOptions), he should provide a non-null logOptions.